gnome/sound-theme-freedesktop/Portfile

```diff
@@ -16,23 +17,10 @@ master_sites        https://people.freedesktop.org/~mccann/dist/
 use_bzip2           yes
 
 checksums           rmd160  68c5bea866c8db3854a4172860bdbb9dc7e4cccc \
-                    sha256  cb518b20eef05ec2e82dda1fa89a292c1760dc023aba91b8aa69bafac85e8a14
+                    sha256  cb518b20eef05ec2e82dda1fa89a292c1760dc023aba91b8aa69bafac85e8a14 \
+                    size    478237
 
-depends_build       port:pkgconfig \
-                    port:intltool \
-                    port:autoconf \
-                    port:automake \
-                    path:bin/glib-gettextize:glib2 \
-                    port:gnome-common
-
-patchfiles          patch-Makefile.am.diff
-
-# reconfigure using autogen.sh for intltool 0.51 compatibility
-
-post-patch {
-    xinstall -m 755 ${filespath}/autogen.sh ${worksrcpath}
-}
-
-configure.cmd       ./autogen.sh
```

Comment:
I think it would be preferable not to remove the Portfile code that regenerates the configure script for intltool 0.51 compatibility. The build log contains this:

```
  /opt/local/bin/intltool-merge  ; /opt/local/bin/intltool-merge ./po index.theme.in index.theme -d -u -c ./po/.intltool-merge-cache
  Usage: intltool-merge [OPTION]... PO_DIRECTORY FILENAME OUTPUT_FILE
  Generates an output file that includes some localized attributes from an
  untranslated source file.
  
  Mandatory options: (exactly one must be specified)
    -b, --ba-style         includes translations in the bonobo-activation style
    -d, --desktop-style    includes translations in the desktop style
    -k, --keys-style       includes translations in the keys style
    -s, --schemas-style    includes translations in the schemas style
    -r, --rfc822deb-style  includes translations in the RFC822 style
        --quoted-style     includes translations in the quoted string style
        --quotedxml-style  includes translations in the quoted xml string style
    -x, --xml-style        includes translations in the standard xml style
  
  Other options:
    -u, --utf8             convert all strings to UTF-8 before merging 
                           (default for everything except RFC822 style)
    -p, --pass-through     deprecated, does nothing and issues a warning
    -m, --multiple-output  output one localized file per locale, instead of 
  	                 a single file containing all localized elements
        --no-translations  do not merge any translations: only generates the
                           unlocalised (English) version -- applies only
                           to XML merging
    -c, --cache=FILE       specify cache file name
                           (usually $top_builddir/po/.intltool-merge-cache)
    -q, --quiet            suppress most messages
        --help             display this help and exit
        --version          output version information and exit
  
  Report bugs to http://bugs.launchpad.net/intltool
  Generating and caching the translation database
  Merging translations into index.theme.
```

This didn't happen before your change.

I compared the destroot contents before and after your change and there was no difference (which suggests the revision should not be increased) but I would still err on the side of reducing weird messages in the log. autoreconfing to be compatible with intltool 0.51 might also prevent future problems if intltool is updated to a newer version.

Reply:
It doesn't seem like there's an actual issue with intltool compatibility, it's possible this was added as part of a mass change ([relevant ticket](https://trac.macports.org/ticket/47159)). As for the message in the log, the Makefile runs intltool-merge without arguments and what's printed is expected.

Reply:
> I think it would be preferable not to remove the Portfile code that regenerates the configure script for intltool 0.51 compatibility. The build log contains this:
> 
> <details>
>   <summary>Expand for Details</summary>
> 
> ```
>   /opt/local/bin/intltool-merge  ; /opt/local/bin/intltool-merge ./po index.theme.in index.theme -d -u -c ./po/.intltool-merge-cache
>   Usage: intltool-merge [OPTION]... PO_DIRECTORY FILENAME OUTPUT_FILE
>   Generates an output file that includes some localized attributes from an
>   untranslated source file.
>   
>   Mandatory options: (exactly one must be specified)
>     -b, --ba-style         includes translations in the bonobo-activation style
>     -d, --desktop-style    includes translations in the desktop style
>     -k, --keys-style       includes translations in the keys style
>     -s, --schemas-style    includes translations in the schemas style
>     -r, --rfc822deb-style  includes translations in the RFC822 style
>         --quoted-style     includes translations in the quoted string style
>         --quotedxml-style  includes translations in the quoted xml string style
>     -x, --xml-style        includes translations in the standard xml style
>   
>   Other options:
>     -u, --utf8             convert all strings to UTF-8 before merging 
>                            (default for everything except RFC822 style)
>     -p, --pass-through     deprecated, does nothing and issues a warning
>     -m, --multiple-output  output one localized file per locale, instead of 
>   	                 a single file containing all localized elements
>         --no-translations  do not merge any translations: only generates the
>                            unlocalised (English) version -- applies only
>                            to XML merging
>     -c, --cache=FILE       specify cache file name
>                            (usually $top_builddir/po/.intltool-merge-cache)
>     -q, --quiet            suppress most messages
>         --help             display this help and exit
>         --version          output version information and exit
>   
>   Report bugs to http://bugs.launchpad.net/intltool
>   Generating and caching the translation database
>   Merging translations into index.theme.
> ```
> 
> </details>
> 
> This didn't happen before your change.
> 
> I compared the destroot contents before and after your change and there was no difference (which suggests the revision should not be increased) but I would still err on the side of reducing weird messages in the log. autoreconfing to be compatible with intltool 0.51 might also prevent future problems if intltool is updated to a newer version.

Agreed. I'm currently testing locally, and I'll see if we can eliminate that.

Reply:
> Mohamed, can you take a closer look at the configure script, to see why `intltool-merge` is being invoked without any arguments?
> 
> ```
> /opt/local/bin/intltool-merge  ; /opt/local/bin/intltool-merge ./po index.theme.in index.theme -d -u -c ./po/.intltool-merge-cache
> ```

P.s. This is still occurring, with your latest commit from this morning.

Reply:
The `%.theme` target in the `Makefile.am` file is invoking `intltool-merge` with a wildcard argument, but the wildcard returns no files. The current "fix" for this was to remove that invocation completely (`patch-Makefile.am.diff`). Since this is pointless as it has no effect on the destroot contents (as @ryandesign noted) and not worth bringing in autoconf/automake for, I removed that patch. It is also erroneous because that wildcard might resolve to some files in the future or some scenario.

Reply:
> The `%.theme` target in the `Makefile.am` file is invoking `intltool-merge` with a wildcard argument, but the wildcard returns no files. The current "fix" for this was to remove that invocation completely (`patch-Makefile.am.diff`). Since this is pointless as it has no effect on the destroot contents (as @ryandesign noted) and not worth bringing in autoconf/automake for, I removed that patch. It is also erroneous because that wildcard might resolve to some files in the future or some scenario.

Makes sense.

So perhaps you can update the makefile to check whether any files exist for said wildcard, and only run the command if there are any?

Reply:
Why though? That seems like a complicated change and I'm not sure what's the benefit. This is the intended Makefile from upstream and it builds without errors.

Reply:
> When a command is run without arguments - such that usage info is printed - that's generally a big red flag that something may not be working properly.

But we know it's working properly.

Adding the patch again is not possible as I also ripped out automake.

Reply:
> > Why though? That seems like a complicated change and I'm not sure what's the benefit. This is the intended Makefile from upstream and it builds without errors.
> 
> Ah, but there is a very important benefit: When a command is run without arguments - such that usage info is printed - that's generally a big red flag that something may not be working properly. And that's why the patch is there.
> 
> As for the complexity argument, there's absolutely none involved: Re-add patch `patch-Makefile.am.diff`, and you're done.
> 
> That still accomplishes your goal of port simplification. And I'll happily merge once you've done that.

Ugh, sorry, we'd have to regenerate everything for that to come into play. (Technically we could also patch `Makefile` via `post-configure`, but that seems hacky.)

I guess I'm fine with it as-is.